src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -0,0 +1,170 @@
+     * placeholders {@code "<SomeParameter>"} such as in
+     * {@code "pluginId.give.item.<ItemType>"}.
+     *
+     * @return The permission id
+     */
+    String getId();
+
+    /**
+     * Gets a short description of the linked permission. May include a link to
+     * a more detailed description on the plugin's web page.
+     *
+     * @return A short description of the linked permission
+     */
+    Text getDescription();
+
+    /**
+     * Gets the human readable <b>suggested</b> rank the players should have
+     * before they should have access to this permission.
+     *
+     * <p><b>Note:</b> This is a suggestion for server owners and does not have
+     * any impact on the results of the permission checks.</p>
+     *
+     * @return The suggested player rank the permission should be granted to
+     */
+    Optional<String> getSuggestedRank();
+
+    /**
+     * Gets the owning plugin the permission belongs to.
+     *
+     * @return The owning plugin the permission belongs to
+     */
+    PluginContainer getOwner();
+
+    /**
+     * A builder for permission descriptions.
+     */
+    interface Builder {
+
+        /**
+         * Sets the permission id for the description this builder creates. May
+         * include placeholders {@code "<SomeParameter>"} such as in
+         * {@code "pluginId.give.item.<ItemType>"}.
+         *
+         * @param permissionId The permission id
+         * @return This builder for chaining
+         */
+        Builder id(String permissionId);
+
+        /**
+         * Sets the short description to use. May include a link to a more
+         * detailed description on the plugin's web page.
+         *
+         * @param description The short description to use
+         * @return This builder for chaining
+         */
+        Builder description(Text description);
+
+        /**
+         * Sets the suggested rank players should have before being able to
+         * access this permission. Setting this is optional although
+         * recommended.
+         *
+         * @param rank The suggested rank to have
+         * @return This builder for chaining
+         */
+        Builder suggestedRank(@Nullable String rank);
+
+        /**
+         * Creates and registers a new {@link PermissionDescription} instance
+         * with the given settings.
+         *
+         * @return The newly created permission description instance
+         * @throws IllegalStateException If there are any settings left unset or
+         *         a description with the given permission id was already
+         *         registered and the {@link PermissionService} does not support
+         *         overwriting descriptions
+         */
+        PermissionDescription register() throws IllegalStateException;
+
+        /**
+         * Resets this builder to its default settings.
+         *
+         * @return This builder for chaining
+         */
+        Builder reset();
```

Comment:
I don't think this is necessary. Builders are used only once, so why would a reset be needed?